driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -11,19 +15,77 @@ use futures::{
 };
 use orderbook::Orderbook;
 use std::future::Future;
-use std::sync::{Arc, Mutex};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex,
+};
+
+/// An event based orderbook that automatically updates itself with new events from the contract.
+#[derive(Debug)]
+pub struct UpdatingOrderbook {
+    orderbook: Arc<Mutex<Orderbook>>,
+    orderbook_ready: Arc<AtomicBool>,
+    // When this struct is dropped this sender will be dropped which makes the updater thread stop.
+    _exit_tx: oneshot::Sender<()>,
+}
+
+impl UpdatingOrderbook {
+    pub fn new(
+        _contract: &impl StableXContract,
+        block_timestamp: impl BlockTimestamp + Send + 'static,
+    ) -> Self {
+        let orderbook = Arc::new(Mutex::new(Orderbook::default()));
+        let orderbook_clone = orderbook.clone();
+        let orderbook_ready = Arc::new(AtomicBool::new(false));
+        let orderbook_ready_clone = orderbook_ready.clone();
```

Comment:
@nlordell wasn't there a syntactically nice way of doing the clone inside the closure to avoid ugly naming like this?